src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -92,180 +96,406 @@ struct NetworkInfo
     NetworkType mNetworkType;
     union NetworkData
     {
-#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
         Thread::OperationalDataset mThread;
-#endif
-#if defined(CHIP_DEVICE_LAYER_TARGET)
         WiFiNetworkInfo mWiFi;
-#endif
     } mData;
 };
 
+enum class ClusterState : uint8_t
+{
+    kIdle            = 0,
+    kEnablingNetwork = 1,
+};
+
+} // namespace Internal
+
 namespace {
 // The internal network info containing credentials. Need to find some better place to save these info.
 NetworkInfo sNetworks[kMaxNetworks];
 } // namespace
 
-EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan operationalDataset,
-                                                                           uint64_t breadcrumb, uint32_t timeoutMs)
+namespace Internal {
+constexpr uint8_t kInvalidNetworkSeq = 0xFF;
+
+// void EncodeAndSendEnableNetworkResponse(CHIP_ERROR err);
+// void SetEnablingNetworkSeq(uint8_t seq);
+// ClusterState GetClusterState();
+// CHIP_ERROR __attribute__((warn_unused_result)) MoveClusterState(ClusterState newState);
+// void OnNetworkEnableStatusChangeCallback(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg);
+// void SetCommissionerSecureSessionHandle(const SecureSessionHandle & handle);
+
+NetworkInfo * GetNetworkBySeq(uint8_t seq);
+
+chip::NodeId networkCommissionerNodeId = chip::kAnyNodeId;
+ClusterState sClusterState             = ClusterState::kIdle;
+uint8_t sCurrentEnablingNetworkSeq     = kInvalidNetworkSeq;
+SecureSessionHandle sCommissionerSecureSessionHandle;
+
+NetworkInfo * GetNetworkBySeq(uint8_t seq)
```

Comment:
Would "seq" be better called "index" both in the function name and in the arg name?